packages/itwin/reports-config-widget/src/widget/components/ReportHorizontalTile.tsx

```diff
@@ -0,0 +1,130 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { useEffect, useState } from "react";
+import React from "react";
+import type BulkExtractor from "./BulkExtractor";
+import { ExtractionStates, ExtractionStatus } from "./ExtractionStatus";
+import type { BeEvent } from "@itwin/core-bentley";
+import { STATUS_CHECK_INTERVAL } from "./Constants";
+import type { Report } from "@itwin/insights-client";
+import { ReportsConfigWidget } from "../../ReportsConfigWidget";
+import {
+  DropdownMenu,
+  IconButton,
+  MenuItem,
+} from "@itwin/itwinui-react";
+import {
+  SvgDelete,
+  SvgEdit,
+  SvgMore,
+} from "@itwin/itwinui-icons-react";
+import { HorizontalTile } from "./HorizontalTile";
+
+export interface ReportHorizontalTileProps {
+  onClickTitle: (e: any) => void;
+  selected: boolean;
+  setSelectedReports: (value: React.SetStateAction<Report[]>) => void;
+  bulkExtractor: BulkExtractor;
+  jobStartEvent: BeEvent<(reportId: string) => void>;
+  report: Report;
+  onClickDelete: () => void;
+  onClickModify: () => void;
+}
+
+export const ReportHorizontalTile = (props: ReportHorizontalTileProps) => {
+  const [jobStarted, setJobStarted] = useState<boolean>(false);
+  const [extractionState, setExtractionState] = useState<ExtractionStates>(ExtractionStates.None);
+
+  if (props.jobStartEvent)
+    props.jobStartEvent.addListener((startedReportId: string) => {
+      if (startedReportId === props.report.id) {
+        setExtractionState(ExtractionStates.Starting);
+        setJobStarted(true);
+      }
+    });
+
+  useEffect(() => {
+    if (jobStarted) {
+      const interval = window.setInterval(async () => {
+        const state = props.bulkExtractor.getState(props.report.id);
+        if (state) {
+          setExtractionState(state);
+          if (state === ExtractionStates.Failed || state === ExtractionStates.Succeeded) {
+            setJobStarted(false);
+          }
+        }
+      }, STATUS_CHECK_INTERVAL);
+      return () => window.clearInterval(interval);
+    }
```

Comment:
If `jobStarted` is false and `interval` is valid - clearInterval.

Reply:
Interval is cleared when jobStarted is false and everytime before starting a new interval.

Reply:
Having one return statement outside of the `if` statement should be enough.

Reply:
return statement has been moved outside of the if statement

Reply:
resolved